doc/development/python-release.md

```diff
@@ -0,0 +1,67 @@
+# Python Release Process
+
+This document describes the general way in which Python packages are released. 
+
+## Prerequistes: 
+
+- Optionally create a new virtual environment for the `bumpversion`  package, and other release tools
+- Install the bumpversion package , `pip install bumpversion`
+- Following git-flow process
+- Running Travis in the repo, i.e. `.travis.yml` exists and has the `deploy` section defined, (see below)
+- bumpversion.cfg exists, example below
+
+```yaml
+[bumpversion]
+current_version = 0.2.16
+commit = True
+tag = True
+
+[bumpversion:file:setup.py]
+search = version='{current_version}'
+replace = version='{new_version}'
+
+[bumpversion:file:squid_py/__init__.py]
+search = __version__ = '{current_version}'
+replace = __version__ = '{new_version}'
+
+[bumpversion:file:docs/source/conf.py]
+search = {current_version}
+replace = {new_version}
+```
+
```

Comment:
The `__init__.py` section is only needed if there's a line in the `__init__.py` file that sets `__version__`

The `docs/source/conf.py` section is only needed if that file exists.

Generally speaking, bumpversion should update all files that list the version. You can find those files by doing a search across the repo for the current version number (e.g. 0.2.16).